src/v3/tutorial/middleware/README.md

```diff
@@ -0,0 +1,87 @@
+---
+description: This tutorial will show you how to use a middleware.
+full_title: Middleware Tutorial
+---
+
+# Middleware Tutorial
+
+This tutorial will show you how to use a middleware. A middleware is used when an application wants to provide a way for other applications to change its behavior.
+
+## Usage
+
+```javascript
+// To add a callback to a middleware group
+core.make('osjs/middleware').add('middleware-group-name', middlewareCallback);
+// OR
+import osjs from 'osjs';
+
+osjs.middleware('middleware-group-name', middlewareCallback);
+
+// To get the list of callbacks in a middleware group
+const middlewareCallbacks = core.make('osjs/middleware').get('middleware-group-name');
+```
+
+## Example
+
+In an application like a virus scanner we need to add a "Scan" option in the context menu of files in the [File Manager application](https://github.com/os-js/osjs-filemanager-application) that opens the scanner application.
+
+![Example](example.png)
+
+The [File Manager application](https://github.com/os-js/osjs-filemanager-application) should
+provide a way for other applications to add to its edit/context menu. So, it uses a middleware group for its edit/context menu.
```

Comment:
I think this should be a bit more generic :thinking: 

```
The [File Manager application](https://github.com/os-js/osjs-filemanager-application) uses middleware to construct the edit (context) menu.

You can add your own callbacks to this middleware if you need to for example launch your own application from this menu with the selected file as an argument.
```

Reply:
I was thinking of replacing the entire text with the two lines I suggested here. But maybe you have some opinions on this ? :blush:

Reply:
No, I'm not very familiar with Github PRs, sorry. Is it fine now?